packages/dds/map/src/directory.ts

```diff
@@ -1528,18 +1626,17 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		callback: (value: unknown, key: string, map: Map<string, unknown>) => void,
 	): void {
 		this.throwIfDisposed();
-		// eslint-disable-next-line unicorn/no-array-for-each
-		this._storage.forEach((value, key) => {
-			callback(value, key, this);
-		});
+		for (const [key, localValue] of this.internalIterator()) {
+			callback((localValue as { value: unknown }).value, key, this.sequencedStorageData);
```

Comment:
The third argument here should probably just be `this` - otherwise this is exposing a private member, and any usage of it would be making raw changes to the sequenced data rather than modifying the directory through the normal appropriate methods.
```suggestion
			callback((localValue as { value: unknown }).value, key, this);
```